### ISSUE TITLE
Javascript blocks should always require new lines and curly braces

### ISSUE THREAD

**Issue by mollykreis** [ni/javascript-styleguide#86]
For consistency, the body of all blocks in javascript should require new lines and curly braces.

eg:
```
if (true) {
    callSomeFunction();
}
```
not
```
if (true) callSomeFunction();
```

Note: This discussion originally started on [this PR](https://ni.visualstudio.com/DevCentral/_git/Skyline/pullrequest/278123?_a=files&path=%2FWeb%2FWorkspaces%2FSystemLinkShared%2Fprojects%2Fsystemlink-lib-angular%2Fsrc%2Fsl-error%2Ferror.http-interceptor.ts&discussionId=2898297).

**Comment by jattasNI** [comment]
We discussed in person and agreed that the [curly](https://eslint.org/docs/rules/curly) rule set to `all` is our best option for now. We would be ok with allowing just the single line `if` syntax but there's no way to allow that without allowing undesirable syntaxes (the `multi` config comes closest).

**Comment by jattasNI** [comment]
I realized while testing the fix for this in #87 that just using the `curly` rule will still allow this syntax:
`if (foo) { doSomething; }`

Our options are:
1. Allow that syntax by only enabling the `curly` rule
2. Set [`'brace-style': ['error', '1tbs', { allowSingleLine: false }]`](https://eslint.org/docs/rules/brace-style) in addition to `curly` which forces the body and closing brace to be on subsequent lines.
3. Only set `'brace-style': ['error', '1tbs', { allowSingleLine: false }]` but NOT curly, which allows single-line statements on the same line if they don't have braces.

```
// Example A. Allowed by 3
if (foo) doSomething();

// Example B. Allowed by 3
if (foo) doSomethingConditionally(); doSomethingUnconditionallyButItLooksConditional();

// Example C. Allowed by 1
if (foo) { doSomething(); }

// Example D. Allowed by 1
if (foo) { doSomething(); doSomethingElse(); }

// Example E. Always disallowed by Airbnb config for non-block-statement-body-position
if (foo)
   doSomething(); 

// Example F. Always allowed
if (foo) {
   doSomething();
} 
```

The original bug report requested 2. That's my vote too because I don't like permitting example B above and find the example C syntax ugly. But if anyone wants to argue for 1 or 3, we can discuss it again. @rajsite @mure @TrevorKarjanis

**Comment by jattasNI** [comment]
We agreed in person to require the body to be on a new line (option 2). @jattasNI to revive the PR to make this happen.
